gen/qr.hoon

```diff
@@ -0,0 +1,359 @@
+  |=  msg=tape
+  ^-  (list @)
+  =/  cod  (msg-to-codewords msg)
+  =/  pol  (codewords-to-polynomial cod)
+  (byte-encode (synthetic-div [pol generator-polynomial]))
+::
+++  add-data
+  |=  [data=(list @) qr=grid]
+  ^-  grid
+  =/  i=@u  0
+  |-
+  ?:  =(i (lent data))
+    qr
+  ?:  =((snag i data) 1)
+    =/  coords  (snag i data-route)
+    %-  tm  :-  [-.coords +.coords]
+    $(i (add 1 i))
+  $(i (add 1 i))
+::
+++  padding
+  |=  char=@tD
+  ^-  (list @)
+  =/  binary-list-flipped  (rip 0 char)
+  =/  binary-list  (flop binary-list-flipped)
+  =/  padded  (zing [(reap (sub 8 (lent binary-list)) 0) binary-list ~])
+  padded
+::
+++  mode-indicator
+  ^-  (list @)
+  [0 1 0 0 ~] :: alphanumeric mode
+::
+++  char-count-indicator
+  |=  l=tape
+  ^-  (list @)
+  %-  padding  (lent l)
+::
+++  byte-encode
+  |=  m=(list @)
+  ^-  (list @)
+  =/  i=@u  0
+  %-  zing
+  |-
+  ?:  =(i (lent m))
+    ~
+  :-  (padding (snag i m))
+  $(i (add 1 i))
+::
+++  add-padding-zeros
+  :: pad a string of bits to ensure it is divisble by 8
+  |=  wip=(list @)
+  ^-  (list @)
+  %-  zing
+  :-  wip
+  :-  (reap (sub (lent wip) (mul 8 (div (lent wip) 8))) 0)   :: make sure this is an evenly divisible by 8
+  ~
+::
+++  add-padding-bytes
+  :: pad a string of bits to ensure it is 864 bits long
+  :: fill the rest with 11101100 and 00010001 repeatedly
+  |=  l=(list @)
+  ^-  (list @)
+  %-  zing
+  :-  l
+  =/  i=@u  (lent l)
+  |-
+  ?:  (lte (add 16 i) 864)
+    :-  [1 1 1 0 1 1 0 0 0 0 0 1 0 0 0 1 ~]
+        $(i (add 16 i))
+  ?:  =(i 856)
+    :-  [1 1 1 0 1 1 0 0 ~]
+        ~
+  ~
+::
+++  msg-to-codewords
+  |=  m=tape
+  ^-  (list @)
+  %-  add-padding-bytes
+  %-  add-padding-zeros
+  %-  zing
+  :-  mode-indicator
+  :-  (char-count-indicator m)
+  :-  (byte-encode m)
+  ~
+::
+++  codewords-to-polynomial
+  ::  chunk a list of bits into bytes, then make a list of integers
+  ::  corresponding to the coefficients of the message polynomial
+  |=  l=(list @)
+  ^-  (list @u)
+  =/  i=@u  0
+  |-
+  ?:  =(i (lent l))
+    ~
+  :-  (rep 0 (flop (scag 8 (slag i l))))
+  $(i (add 8 i))
+::
+++  galois-multiply
+  |=  [x=@u y=@u]
+  ^-  @u
+  ?:  =(x 0)  0
+  ?:  =(y 0)  0
+  (snag (add (snag x galois-logs) (snag y galois-logs)) galois-antilogs)
+::
+++  xor-coefficient
+  |=  [l=(list @u) i=@u x=@u]
+  =/  new  (mix x (snag i l))
+  %-  zing 
+  [(scag i l) [new ~] (slag (add 1 i) l) ~]
+::
+++  synthetic-div
+  :: getting help from here https://en.wikiversity.org/wiki/Reed%E2%80%93Solomon_codes_for_coders#Encoding_main_function
+  |=  [data=(list @u) gen=(list @u)]
+  ^-  (list @u)
+  =/  i=@u  0
+  =/  j=@u  1
+  =/  out=(list @u)  (zing [data (reap (sub (lent gen) 1) 0) ~])
+  |-
+  ?:  =(i (lent data))
+    (zing [data (slag (lent data) out) ~])
+  ?:  =((snag i out) 0)
+    $(i (add 1 i), j 1)
+  ?:  =(j (lent gen))
+    $(i (add 1 i), j 1)
+  =/  update  (xor-coefficient [out (add i j) (galois-multiply (snag j gen) (snag i out))])
```

Comment:
Using `=/` makes it easy to split the computation out over multiple lines.

```hoon
=/  update
  %^  xor-coefficient  out
    (add i j)
  %+  galois-multiply
    (snag j gen)
  (snag i out)
etc
```

Since your `etc` is really small here, it might be worth using `=;` instead. As shown in another comment:

```hoon
=;  update  etc
%^  xor-coefficient  out
same-as-above
```